src/charm.py

```diff
@@ -360,7 +367,7 @@ def _on_peer_relation_joined(self, event: RelationJoinedEvent):
 
         except MySQLAddInstanceToClusterError:
             logger.debug(f"Unable to add instance {new_instance_fqdn} to cluster.")
-            event.defer()
```

Comment:
you mentioned this was an issue in the PR description, why? Wouldn't we want to try again?

Reply:
this i thought was a problem, because rarely does adding an instance to a cluster twice resolve the original issue (plus the instance may have some stale bad state). if we defer, we never surface the problem to the admin, which is why i think it better to move to blocked status